File: online-shop/online-shop-backend/src/main/java/com/example/onlineshop/security/models/OrderProduct.java
package com.example.onlineshop.security.models;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;

@Setter
@Getter
@Entity
@RequiredArgsConstructor
public class OrderProduct {

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private Long id;

    @ManyToOne
    private BillingData billingData;

    @ManyToOne
    private Product product;

    @ManyToOne
    private User client;

    @Column(name = "quantity")
    private Integer quantity;

    @Column(name="finalized")
    private Boolean finalized;

    @Column(name="orderDate")
    private Date orderDate;

}

File: online-shop/online-shop-frontend/src/app/models/orderProduct.ts
import {Product} from "./product";
import {User} from "./user";
import {BillingData} from "./billingData";

export class OrderProduct {

  id: number = 0;
  billingData: BillingData = new BillingData();
  product:Product = new Product();
  client:User =new User();
  finalized:boolean =false;
  quantity:number=0;
  orderDate: string = '';


  constructor(){
  }

}
